src/gateway.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <golioth/gateway.h>
+#include "coap_blockwise.h"
+
+#if defined(CONFIG_GOLIOTH_GATEWAY)
```

Comment:
I lack context here, really, but how come we're doing this throughout instead of zephyr's cmake utils?

```cmake
zephyr_library_sources_ifdef(CONFIG_GOLIOTH_GATEWAY, src/gateway.c)
```